department-of-veterans-affairs/caseflow
# Refactor Schedule Veterans page to fetch hearings separately

Resolves [CASEFLOW-1755](https://vajira.max.gov/browse/CASEFLOW-1755)

**PR Stack**

[Part 1](https://github.com/department-of-veterans-affairs/caseflow/pull/17005)
Part 2

### Description
Frontend changes to retrieve hearings for a Hearing Day only when clicked on, rather than fetching them all on page load. Also refactors to use the `filledSlots` property, since there's no longer a hearings array length to provide the number of filled slots. 

### Testing Plan
1. As the BVASYELLOW user, go to http://localhost:3000/hearings/schedule/assign and select regional office of St Petersburg, FL. 
2. Verify that the `3 of 10 scheduled` counts in the list on the left-hand side are accurate, and that the "slots remaining" heading is accurate (see screenshot below). 
3. Verify that the first Hearing Day's hearings are loaded when the page loads.
4. Verify that clicking on other Hearing Days results in a loading indicator and then shows hearings for that day (see gif below).  

### User Facing Changes
![Screen Shot 2021-12-01 at 10 13 47 AM](https://user-images.githubusercontent.com/92495948/144291443-9c9d623f-1fae-4a4e-83b1-c4ffa2f2a3ef.png)  

![Loading_while_fetching_hearings](https://user-images.githubusercontent.com/92495948/144291480-a27ae40b-9aeb-4596-ac1f-99a626582ad0.gif)
